dot/node_integration_test.go

```diff
@@ -9,24 +9,197 @@ import (
 	"context"
 	"encoding/hex"
 	"encoding/json"
+	"fmt"
 	"os"
 	"path/filepath"
 	"testing"
 
 	"github.com/ChainSafe/gossamer/dot/core"
+	digest "github.com/ChainSafe/gossamer/dot/digest"
+	network "github.com/ChainSafe/gossamer/dot/network"
 	"github.com/ChainSafe/gossamer/dot/state"
+	dotsync "github.com/ChainSafe/gossamer/dot/sync"
+	system "github.com/ChainSafe/gossamer/dot/system"
+	"github.com/ChainSafe/gossamer/dot/telemetry"
+	"github.com/ChainSafe/gossamer/dot/types"
+	"github.com/ChainSafe/gossamer/internal/log"
 	"github.com/ChainSafe/gossamer/lib/babe"
 	"github.com/ChainSafe/gossamer/lib/common"
 	"github.com/ChainSafe/gossamer/lib/crypto/sr25519"
 	"github.com/ChainSafe/gossamer/lib/genesis"
 	"github.com/ChainSafe/gossamer/lib/grandpa"
 	"github.com/ChainSafe/gossamer/lib/keystore"
 	"github.com/ChainSafe/gossamer/lib/runtime"
+	"github.com/ChainSafe/gossamer/lib/runtime/wasmer"
 	"github.com/ChainSafe/gossamer/lib/trie"
 	"github.com/ChainSafe/gossamer/lib/utils"
+	gomock "github.com/golang/mock/gomock"
+	"github.com/stretchr/testify/assert"
 	"github.com/stretchr/testify/require"
 )
 
+func TestNewNode(t *testing.T) {
+	ctrl := gomock.NewController(t)
+
+	mockTelemetryClient := NewMockClient(ctrl)
+	mockTelemetryClient.EXPECT().SendMessage(gomock.Any())
+
+	initConfig := NewTestConfig(t)
+
+	genFile := NewTestGenesisRawFile(t, initConfig)
+
+	networkConfig := &network.Config{
+		BasePath:    t.TempDir(),
+		NoBootstrap: true,
+		NoMDNS:      true,
+	}
+	setConfigTestDefaults(t, networkConfig)
+
+	testNetworkService, err := network.NewService(networkConfig)
+	require.NoError(t, err)
+
+	config := state.Config{
+		Path:     initConfig.Global.BasePath,
+		LogLevel: initConfig.Log.StateLvl,
+	}
+
+	dotConfig := &Config{
+		Global:  GlobalConfig{BasePath: initConfig.Global.BasePath},
+		Init:    InitConfig{Genesis: genFile},
+		Account: AccountConfig{Key: "alice"},
+		Core: CoreConfig{
+			Roles:           common.FullNodeRole,
+			WasmInterpreter: wasmer.Name,
+		},
+	}
+
+	dotConfig.Init = InitConfig{Genesis: genFile}
+	dotConfig.Account = AccountConfig{Key: "alice"}
+	dotConfig.Core.Roles = common.FullNodeRole
+	dotConfig.Core.WasmInterpreter = wasmer.Name
+	dotConfig.Global.Name = "TestNode"
+
+	ks, err := initKeystore(t, dotConfig)
+	assert.NoError(t, err)
+
+	mockServiceRegistry := NewMockServiceRegisterer(ctrl)
+	mockServiceRegistry.EXPECT().RegisterService(gomock.Any()).Times(8)
+
+	m := NewMocknodeBuilderIface(ctrl)
```

Comment:
To make sure i understand, this is a temp PR to speed up CI, but in future will be more intentional about where individual tests are?

Reply:
This PR renames entire files so git blame information is kept as much as possible. After this PR (or never), we could gradually fish out fast tests from these integration test files and move them to unit test files, although that would also lose git blame information :cry: